java/fory-core/src/main/java/org/apache/fory/codegen/Expression.java

```diff
@@ -2287,7 +2288,8 @@ public ExprCode doGenCode(CodegenContext ctx) {
       String i = ctx.newName("i");
       String elemValue = ctx.newName("elemValue");
       Expression elementExpr =
-          action.apply(new Reference(i), new Reference(elemValue, elementType, false));
+          action.apply(
+              new Reference(i), new Reference(elemValue, elementType, !elementType.isPrimitive()));
```

Comment:
Could we add an `elementNullable` field and pass it when creating `ForEach` esxpression? And condition that `elementNullable` here?

For `java/fory-core/src/main/java/org/apache/fory/builder/BaseObjectCodecBuilder#writeContainerElements`, we don't need `nullable` variable since we check and computed nullability ahead in `writeElementsHeader`